src/main/java/jenkins/plugin/randomjobbuilder/GeneratorController.java

```diff
@@ -0,0 +1,365 @@
+                runs.add(run);
+            } else {
+                HashSet<Run> runCollection = new HashSet<>();
+                runCollection.add(run);
+                generatorToRuns.put(generator.getGeneratorId(), runCollection);
+            }
+        }
+    }
+
+    /**
+     * Remove a run (as if completed) that was tracked against a given generator
+     * @param generator
+     * @param run
+     */
+    public void removeRun(@Nonnull LoadGenerator generator, Run run) {
+        synchronized (generator) {
+            Collection<Run> runs = generatorToRuns.get(generator.getGeneratorId());
+            if (runs != null) {
+                runs.remove(run);
+            }
+        }
+    }
+
+    /**
+     * Get number of tracked queued items for generator
+     * @param gen
+     * @return Queued count
+     */
+    public int getQueuedCount(@Nonnull LoadGenerator gen) {
+        synchronized (gen) {
+            AtomicInteger val = queueTaskCount.get(gen.getGeneratorId());
+            int count = (val != null) ? val.get() : 0;
+            return count;
+        }
+    }
+
+    /**
+     * Get the count of actually running jobs for the generator
+     * @param gen
+     * @return
+     */
+    public int getRunningCount(@Nonnull LoadGenerator gen) {
+        synchronized (gen) {
+            Collection<Run> runColl = generatorToRuns.get(gen.getGeneratorId());
+            return (runColl != null) ? runColl.size() : 0;
+        }
+    }
+
+    /** Sum of both queued and actively running runs for the given generator
+     * @param gen
+     */
+    int getQueuedAndRunningCount(@Nonnull LoadGenerator gen) {
+        synchronized (gen) {
+            AtomicInteger val = queueTaskCount.get(gen.getGeneratorId());
+            int count = (val != null) ? val.get() : 0;
+            Collection<Run> runColl = generatorToRuns.get(gen.getGeneratorId());
+            if (runColl != null) {
+                count += runColl.size();
+            }
+            return count;
+        }
+    }
+
+    /** Returns a snapshot of current tracked runs for generator
+     *  @param generator
+     */
+    @Nonnull
+    private Collection<Run> getRuns(@Nonnull LoadGenerator generator) {
+        synchronized (generator) {
+            Collection<Run> runs = generatorToRuns.get(generator.getGeneratorId());
+            return (runs != null && runs.size() > 0) ? new ArrayList<Run>(runs) : Collections.<Run>emptySet();
+        }
+    }
+
+    /** Triggers runs as requested by the LoadGenerator for a REGISTERED generator only
+     *  @param gen
+     *  @return Number of runs triggered as a result of load check
+     */
+    public int checkLoadAndTriggerRuns(@Nonnull LoadGenerator gen) {
+        LoadGenerator registeredGen = getRegisteredGeneratorbyId(gen.generatorId);
+        if (registeredGen == null) {
+            // Not a registered generator, bomb out
+            return 0;
+        }
+        synchronized (registeredGen) {
```

Comment:
This delegates heavily to the LoadGenerator so we can extend that to customize how many jobs it actually runs (ramp up/ramp down and adding some randomization in timing)